Assets/Mirror/Components/NetworkTransformBase.cs

```diff
@@ -16,131 +16,265 @@
 // * Only way for smooth movement is to use a fixed movement speed during
 //   interpolation. interpolation over time is never that good.
 //
+using System;
 using UnityEngine;
 
 namespace Mirror
 {
     public abstract class NetworkTransformBase : NetworkBehaviour
     {
+        // target transform to sync. can be on a child.
+        protected abstract Transform targetTransform { get; }
+
         [Header("Authority")]
-        [Tooltip("Set to true if moves come from owner client, set to false if moves always come from server")]
+
+        [Tooltip("Set to true if updates come from owner client, set to false if updates always come from server")]
+        [SyncVar]
         public bool clientAuthority;
 
-        /// <summary>
-        /// We need to store this locally on the server so clients can't request Authority when ever they like
-        /// </summary>
-        bool clientAuthorityBeforeTeleport;
+        [Header("Application of Updates")]
 
-        // Is this a client with authority over this transform?
-        // This component could be on the player object or any object that has been assigned authority to this client.
-        bool IsClientWithAuthority => hasAuthority && clientAuthority;
+        [Tooltip("Set to true if position changes should be applied")]
+        [SyncVar]
+        public bool applyPosition = true;
+
+        [Tooltip("Set to true if rotation changes should be applied")]
+        [SyncVar]
+        public bool applyRotation = true;
+
+        [Tooltip("Set to true if scale changes should be applied")]
+        [SyncVar]
+        public bool applyScale = true;
+
+        [Header("Interpolation")]
+
+        [Tooltip("Set to true if position should be interpolated")]
+        [SyncVar]
+        public bool interpolatePosition = true;
```

Comment:
why would pos/rot/scale ever not be interpolated?

Reply:
Resetting character to spawn point after death, or any other time you want the object to snap to the new position.  Tetris blocks don't interpolate as they fall or rotate, for example.